src/fhirtypes/ElementDefinition.ts

```diff
@@ -1104,13 +1104,20 @@ export class ElementDefinition {
         const found = this.patternReference;
         const foundFshReference = new FshReference(found.reference, found.display);
         // Check if the new quantity matches the current
-        if (!value.equals(foundFshReference)) {
+        if (
+          !value.equals(foundFshReference) &&
+          !(foundFshReference.display == null && value.display != null)
+        ) {
```

Comment:
I don't think this is quite right because now this would not throw an error, right?
```
* subject = Reference(foo)
* subject = Reference(bar) "Bar"
```
... because the above `if` statement would essentially be evaluating `if (true and false)`.

Reply:
Maybe you should just update the `equals` function to have an `ignoreAddedDisplay` argument (or something).